### DISCUSSION TITLE
Debug Build windows

### DISCUSSION THREAD

**Discussion by skhiat** [halide/Halide#5902] [Q&A]
Hello,

I try to have a Debug build locally with Visual Studio, to have Release and Debug symbol/pdb. Currently I have various:

```
cl : command line error D8016: '/O2' and '/RTC1' command-line options are incompatible [F:\git\Halide\BuildVS\test\Halide_expect_abort.vcxproj]
cl : command line error D8016: '/O2' and '/RTC1' command-line options are incompatible [F:\git\Halide\BuildVS\test\generator\acquire_release.generator.vcxproj]
...
```

I didn't find the related custom build. I try to add blindly:

```
STRING (REGEX REPLACE "/RTC(su|[1su])" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

To remove runtime checks. Do you have an idea which custom build it's from?

Those flag seem to be added by default by CMake on Visual Studio solution generation.

I used vcpkg for dependencies, so I used LLVM 10.0.0#1 with {aarch64, amdgup, arm, hexagon, mips, nvptx, powerpc, riscv, x86, web-assembly}, I don't think it's related.

Command Line:

```
mkdir BuildVS
cd BuildVS
cmake .. -DCMAKE_TOOLCHAIN_FILE=F:/git/vcpkg/scripts/buildsystems/vcpkg.cmake
cmake --build . -DCMAKE_TOOLCHAIN_FILE=F:/git/vcpkg/scripts/buildsystems/vcpkg.cmake
```

**Comment by alexreinking** [comment]
It looks like vcpkg (not CMake itself) adds the `/RTC1` flag to `CMAKE_CXX_FLAGS_DEBUG` (not `CMAKE_CXX_FLAGS`) by default. I see this in `scripts/toolchains/windows.cmake`. However, I do not know where `/O2` comes from. We don't add it anywhere in our build and vcpkg only adds it to `CMAKE_CXX_FLAGS_RELEASE`.

I wonder if LLVM is injecting it via `LLVM_DEFINITIONS`. It might just fix things to update vcpkg and upgrade LLVM to version 11 (through vcpkg).

**Reply by skhiat** [reply]
[Fixed]
By forcing by hand reconfiguring:
cmake .. -DCMAKE_C_FLAGS_DEBUG="/MDd /Zi /Ob0 /Od" -DCMAKE_CXX_FLAGS_DEBUG="/MDd /Zi /Ob0 /Od" -DCMAKE_TOOLCHAIN_FILE=F:/git/vcpkg/scripts/buildsystems/vcpkg.cmake

cmake --build . -DCMAKE_TOOLCHAIN_FILE=F:/git/vcpkg/scripts/buildsystems/vcpkg.cmake

I tried to update LLVM to 11.1.0 with the same issue.
And nothing special on my file:
https://github.com/halide/Halide/commit/71b895e916944c57e267f0968f5b1770da359112
